prog/vm/github_runner.rb

```diff
@@ -287,22 +287,31 @@ def setup_info
 
     hop_wait
   rescue Octokit::Conflict => e
-    unless e.message.include?("Already exists")
-      raise e
-    end
+    raise e unless e.message.include?("Already exists")
+
     # If the runner already exists at GitHub side, this suggests that the
-    # process terminated prematurely before start the runner script and hop wait.
-    # We need to locate the 'runner_id' using the name and delete it.
-    # After this, we can register the runner again.
+    # process terminated prematurely before hop wait. We can't be sure if the
+    # script was started or not without checking the runner status. We need to
+    # locate the runner using the name and decide delete or continue to wait.
     runners = github_client.paginate("/repos/#{github_runner.repository_name}/actions/runners") do |data, last_response|
       data[:runners].concat last_response.data[:runners]
     end
     unless (runner = runners[:runners].find { _1[:name] == github_runner.ubid.to_s })
       fail "BUG: Failed with runner already exists error but couldn't find it"
     end
-    Clog.emit("Deleting GithubRunner because it already exists") { {github_runner: github_runner.values.merge({runner_id: runner[:id]})} }
-    github_client.delete("/repos/#{github_runner.repository_name}/actions/runners/#{runner[:id]}")
-    nap 5
+
+    # If the runner script is not started yet, we can delete the runner and
+    # register it again.
+    if vm.sshable.cmd("systemctl show -p SubState --value #{SERVICE_NAME}").chomp == "dead"
```

Comment:
I'm a little puzzled why SERVICE_NAME is outlined. It's referenced rarely, and only from tightly coupled parts of the program. It's very inelastic in how it's likely to ever be changed (right?), to get a feel for things I have to toggle to the test or dereference the symbol and do a bit of mental computation. I have to dereference it to check quoting (since every expansion without a shellescape has to be checked to be thorough, though a constant is a very unlikely source of problem or injection). The cost reduces my reflexes in knowing what to expect when typing "systemctl status". I suppose it helps finds references to the same logical idea, to inline it makes that more inexpensive indeed.

I would call something even smaller than a nit, but I'm wondering what your design sensibility was there. More an intellectual question.

Reply:
There's no special reason. I wished to convert redundant strings into constants. It's rarely referenced, and the constant makes it harder to read. @byucesoy  gave similar feedback before. Let's refactor: https://github.com/ubicloud/ubicloud/pull/1419